org/postgresql/jdbc2/TypeInfoCache.java

```diff
@@ -291,7 +286,8 @@ private PreparedStatement getOidStatement(String pgTypeName) throws SQLException
                 _getOidStatementSimple = _conn.prepareStatement(sql);
             }
             //default arrays are represented with _ as prefix ... this dont even work for public schema fully
-            _getOidStatementSimple.setString(1, isArray ? "_" + pgTypeName.substring(0, pgTypeName.length() - 2) : pgTypeName);
+            String lcName = pgTypeName.toLowerCase(Locale.ENGLISH);
```

Comment:
I actually don't think this is a good idea, but if you want such behavior probably a better way to go about it is to register built-in types and aliases as upercase.